components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -145,8 +148,9 @@ open class GleanInternalAPI internal constructor () {
         @Suppress("EXPERIMENTAL_API_USAGE")
         Dispatchers.API.flushQueuedInitialTasks()
 
-        // At this point, all metrics and events can be recorded.
-        ProcessLifecycleOwner.get().lifecycle.addObserver(gleanLifecycleObserver)
+        CoroutineScope(KotlinxDispatchers.Main).launch {
```

Comment:
Any reason here you aren't using [MainScope](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-main-scope.html)?

As in:
```Kotlin
MainScope().launch {
    ...
}
```

It seems a little clearer/cleaner...

Reply:
Ah -- I just didn't know about it.  I agree, that's clearer.